docs/docs/tutorial/chapter7/rbac.md

```diff
@@ -96,6 +96,74 @@ export const getCurrentUser = async (session) => {
 </TabItem>
 </Tabs>
 
+
+
+<ShowForTs>
+
+### Fixing the hasRole function
+
+At this point, you might notice an error in your src/lib/auth.ts file, in the `hasRole` function. TypeScript is trying to help you here, by highlighting that roles can never be an an array of strings.
+
+
+```ts
+export const hasRole = (roles: AllowedRoles): boolean => {
+  /*
+  ...
+  ...
+  */
+    } else if (Array.isArray(currentUserRoles)) {
+
+      // 👇 TypeScript will now be telling you 'some' does not exist on type never
+      // highlight-next-line
+      return currentUserRoles?.some((allowedRole) => roles === allowedRole)
+    }
+  }
+ ```
+
+This is because, we now know that the type of your `currentUser.roles` is a `string` - based on the type being returned from Prisma. So you can safely remove the block of code where its checking if roles is an array.
+
+
+```diff title="api/src/lib/auth.ts"
+export const hasRole = (roles: AllowedRoles): boolean => {
+  if (!isAuthenticated()) {
+    return false
+  }
+
+  const currentUserRoles = context.currentUser?.roles
+
+  if (typeof roles === 'string') {
+    if (typeof currentUserRoles === 'string') {
```

Comment:
Do we need this if statement if we know `currentUserRoles` is a string?

Reply:
Yeah - so we definitely don't need this code - and my first plan was to just copy paste these changes https://redwoodjs.com/docs/canary/typescript/strict-mode#b-roles-on-current-user-is-a-string - but it really does make it seem scary. 

So to keep it simple, I'm only suggesting the bare minimum to make the red squigglies go away - which seems to be the main problem that some people report. Often people aren't reading the code - just get scared when they see the red.